apps/system/utils/fscmd.c

```diff
@@ -1084,10 +1084,26 @@ static int tash_rm(int argc, char **args)
 {
 	char *fullpath;
 	int ret = OK;
+	struct stat st;
 
 	if (args[2]) {
 		fullpath = get_fullpath(args[2]);
 		if (strncmp(args[1], "-r", strlen(args[1])) == 0) {
+			/* The request is to delete an entire directory recursively
+			   Check if the entry requested is not a directory and report
+			 */
+			if (stat(fullpath, &st) < 0) {
+				FSCMD_OUTPUT("stat failed with %s\n", fullpath);
+				ret = ERROR;
+				goto errout;
+			}
+
+			if (!S_ISDIR(st.st_mode)) {
```

Comment:
In this case, is `unlink` not needed?
In `delete_entry()`, there is an unlink with same case.

Reply:
This case checks if the path given along with rm -r does not represent a directory, then the command is invalid.